File: CS/DXApplication5/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using DevExpress.XtraEditors;
using DevExpress.XtraScheduler;
using System.Diagnostics;


namespace DXApplication5
{
    public partial class Form1 : XtraForm
    {   
        public Form1()
        {
            InitializeComponent();

            schedulerControl.Start = System.DateTime.Now;
            schedulerControl.GroupType = SchedulerGroupType.Resource;

            InitResources();
            InitAppointments();
        }

        BindingList<CustomResource> resources = new BindingList<CustomResource>();
        BindingList<CustomAppointment> eventList = new BindingList<CustomAppointment>();
        private void InitResources()
        {
            ResourceMappingInfo mappings = this.schedulerStorage.Resources.Mappings;
            mappings.Caption = "Name";
            mappings.Id = "ResourceID";
            
            DataHelper.FillResources(resources, 5);
            this.schedulerStorage.Resources.DataSource = resources;
        }
        void InitAppointments()
        {
            AppointmentMappingInfo mappings = this.schedulerStorage.Appointments.Mappings;
            mappings.Start = "StartTime";
            mappings.End = "EndTime";
            mappings.Subject = "Subject";
            mappings.AllDay = "AllDay";
            mappings.Description = "Description";
            mappings.Label = "Label";
            mappings.Location = "Location";
            mappings.RecurrenceInfo = "RecurrenceInfo";
            mappings.ReminderInfo = "ReminderInfo";
            mappings.ResourceId = "OwnerId";
            mappings.Status = "Status";
            mappings.Type = "EventType";

            
            schedulerStorage.Appointments.CustomFieldMappings.Add(new AppointmentCustomFieldMapping("Priority", "Priority"));
            DataHelper.GenerateEvents(eventList, schedulerStorage.Resources.Count, resources);
            this.schedulerStorage.Appointments.DataSource = eventList;
            
        }
        void Storage_AppointmentsChanged(object sender, DevExpress.XtraScheduler.PersistentObjectsEventArgs e)
        {
            AppointmentBaseCollection appointments = e.Objects as AppointmentBaseCollection;
            foreach (Appointment appointment in appointments)
            {
                Debug.WriteLine(appointment.Subject);
            }
        }
        #region #InplaceEditorShowing
        private void schedulerControl_InplaceEditorShowing(object sender, DevExpress.XtraScheduler.InplaceEditorEventArgs e)
        {
            e.InplaceEditorEx = new MyInplaceEditor(e.SchedulerInplaceEditorEventArgs);
        }
        #endregion #InplaceEditorShowing
    }
}

File: VB/DXApplication5/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Linq
Imports DevExpress.XtraEditors
Imports DevExpress.XtraScheduler
Imports System.Diagnostics


Namespace DXApplication5
    Partial Public Class Form1
        Inherits XtraForm

        Public Sub New()
            InitializeComponent()

            schedulerControl.Start = Date.Now
            schedulerControl.GroupType = SchedulerGroupType.Resource

            InitResources()
            InitAppointments()
        End Sub

        Private resources As New BindingList(Of CustomResource)()
        Private eventList As New BindingList(Of CustomAppointment)()
        Private Sub InitResources()
            Dim mappings As ResourceMappingInfo = Me.schedulerStorage.Resources.Mappings
            mappings.Caption = "Name"
            mappings.Id = "ResourceID"

            DataHelper.FillResources(resources, 5)
            Me.schedulerStorage.Resources.DataSource = resources
        End Sub
        Private Sub InitAppointments()
            Dim mappings As AppointmentMappingInfo = Me.schedulerStorage.Appointments.Mappings
            mappings.Start = "StartTime"
            mappings.End = "EndTime"
            mappings.Subject = "Subject"
            mappings.AllDay = "AllDay"
            mappings.Description = "Description"
            mappings.Label = "Label"
            mappings.Location = "Location"
            mappings.RecurrenceInfo = "RecurrenceInfo"
            mappings.ReminderInfo = "ReminderInfo"
            mappings.ResourceId = "OwnerId"
            mappings.Status = "Status"
            mappings.Type = "EventType"


            schedulerStorage.Appointments.CustomFieldMappings.Add(New AppointmentCustomFieldMapping("Priority", "Priority"))
            DataHelper.GenerateEvents(eventList, schedulerStorage.Resources.Count, resources)
            Me.schedulerStorage.Appointments.DataSource = eventList

        End Sub
        Private Sub Storage_AppointmentsChanged(ByVal sender As Object, ByVal e As DevExpress.XtraScheduler.PersistentObjectsEventArgs) Handles schedulerStorage.AppointmentsChanged
            Dim appointments As AppointmentBaseCollection = TryCast(e.Objects, AppointmentBaseCollection)
            For Each appointment As Appointment In appointments
                Debug.WriteLine(appointment.Subject)
            Next appointment
        End Sub
        #Region "#InplaceEditorShowing"
        Private Sub schedulerControl_InplaceEditorShowing(ByVal sender As Object, ByVal e As DevExpress.XtraScheduler.InplaceEditorEventArgs) Handles schedulerControl.InplaceEditorShowing
            e.InplaceEditorEx = New MyInplaceEditor(e.SchedulerInplaceEditorEventArgs, e.Appointment)
        End Sub
        #End Region ' #InplaceEditorShowing
    End Class
End Namespace